src/txn/txn_recover.c

```diff
@@ -685,7 +685,38 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 
 done:
     WT_ERR(__recovery_set_checkpoint_timestamp(&r));
-    if (do_checkpoint)
+
+    /*
+     * Perform rollback to stable when the connection is not read only and recovery timestamp is
+     * available.
+     */
+    if (!F_ISSET(conn, WT_CONN_READONLY) && conn->txn_global.recovery_timestamp != WT_TS_NONE) {
+        /* Start the eviction threads for rollback to stable if not already started. */
+        if (!eviction_started) {
+            WT_ERR(__wt_evict_create(session));
+            eviction_started = true;
+        }
+
+        /*
+         * Currently, rollback to stable don't do any changes to the tables with no timestamp. In
+         * future to enable rollback to stable on these tables, based on the page write generation
+         * number, the rollback to stable should decide the pages that needs rollback.
+         *
+         * For example, A page with write generation 10 and txnid 20 is updated in one checkpoint
+         * and in the next restart, a new page with write generation 30 and txnid 20 is modified.
+         * The rollback to stable operation should only rollback the latest page changes only based
+         * on the write generation numbers.
```

Comment:
I put this in grammarly, and ended up with:

> Currently, rollback to stable only needs to make changes to tables that use timestamps. That is because eviction does not run in parallel with a checkpoint, so content that is written never uses transaction IDs newer than the checkpoint's transaction ID and thus never needs to be rolled back.
> Once eviction is allowed while a checkpoint is active, it will be necessary to take the page write generation number into account during rollback to stable.
> For example, a page with write generation 10 and txnid 20 is written in one checkpoint, and in the next restart a new page with write generation 30 and txnid 20 is written. The rollback to stable operation should only rollback the latest page changes solely based on the write generation numbers.

Reply:
Fixed.